adapters/smaato/smaato.go

```diff
@@ -365,3 +452,67 @@ func setBannerDimension(banner *openrtb2.Banner) (*openrtb2.Banner, error) {
 
 	return &bannerCopy, nil
 }
+
+func groupImpressionsByPod(imps []openrtb2.Imp) (map[string]([]openrtb2.Imp), []string) {
+	pods := make(map[string][]openrtb2.Imp)
+	orderKeys := make([]string, 0)
+
+	for _, imp := range imps {
+		pod := strings.Split(imp.ID, "_")[0]
+		_, present := pods[pod]
+		pods[pod] = append(pods[pod], imp)
+		if !present {
+			orderKeys = append(orderKeys, pod)
+		}
+	}
+	return pods, orderKeys
+}
+
+//func groupImpressionsByPod(imps []openrtb2.Imp) map[string]([]openrtb2.Imp) {
+//
+//	pods := make(map[string][]openrtb2.Imp)
+//	for _, imp := range imps {
+//		pod := strings.Split(imp.ID, "_")[0]
+//		pods[pod] = append(pods[pod], imp)
+//	}
+//	return pods
+//}
+
+func filterPodImpressions(imps []openrtb2.Imp) ([]openrtb2.Imp, []error) {
+	var n int
+	errors := make([]error, 0, len(imps))
+	for _, imp := range imps {
+		if imp.Video == nil {
+			errors = append(errors, &errortypes.BadInput{Message: "Invalid MediaType. Smaato only supports Video for AdPod."})
+			continue
+		}
+		imps[n] = imp
```

Comment:
inline `n++` maybe?

Reply:
no quite,
n++ is a statement in go and doesn't have a value